libraries/platforms/dromajo-vcs/software/src/args.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * args.c
+ * This file includes routines that queries the arguments from the x86 host and 
+ * loads them into corresponding registers so that BlackParrot can access them
+ */
+
+#include <bp_utils.h>
+#include <bsg_manycore.h>
+#include <bsg_manycore_packet.h>
+#include <bsg_manycore_printing.h>
+#include <stdio.h>
+
+// Number of arguments that can be retrieved is currently based on the size of the buffer.
+// Other ideas (TODO):
+// 1. Create pointers to locations of arguments on the stack (traditional, preferred method)
+// 2. Flush the argument buffer periodically
+int _argc;
+char **_argv;
+static int arg_index = 0;
+static int char_index = 0;
+
+/**
+ * Retrieve arguments from the manycore host
+ */
+void __init_args(void) {
+  int err;
+  hb_mc_packet_t args_req_pkt, args_resp_pkt;
+
+  // Flat buffer to capture all the arguments contiguously
+  char buffer[3000];
+  char *bufptr = buffer;
+  
+  // Create a packet for the x86 host
+  // Host is at POD (X, Y) = (0, 1), SUBCOORD (X, Y) = (0, 0)
+  // BlackParrot FIFO interface is at POD (X, Y) = (0, 1), SUBCOORD (X, Y) = (0, 1)
+  args_req_pkt.request.x_dst = (0 << 4) | 0;
```

Comment:
@drichmond does HOST_X_CORD/HOST_Y_CORD exist in the code, if not can we add it?

Reply:
You can, and should read it from the rom

Reply:
Huh, I currently rely on sending a packet to the host to be able to read the ROM. Any suggestions on how I can send a packet to the host before I can get the coordinates of the host would be helpful haha. Or, if I have to redo the way the ROM is being read, that would also be fine.

Reply:
it doesn't need to be read dynamically, can just set it when compiling the C code

Reply:
I think I'm still confused on why there are so many FIFOs. Why do we need to send packets before reading the ROM?

Reply:
It’s the same interface as x86?

Reply:
In x86 we don't send packets to read the ROM. We do MMIO reads/writes in physical hardware or we use a DPI rom in simulation.

Reply:
Let's set up a time to chat. My recent comments reflect that there's something I'm missing.

Reply:
HOST coords are now fetched from the makefile flow.